pkgs/applications/networking/browsers/firefox-pmos-mobile/default.nix

```diff
@@ -0,0 +1,61 @@
+{ browser
+, lib
+, fetchFromGitLab
+, stdenv
+, wrapFirefox
+, firefoxLibName ? "firefox"
+, forceWayland ? true
+, keepHomepage ? true
+}:
+let
+  version = "2.0.1";
+
+  mobile-config-firefox = stdenv.mkDerivation {
+    name = "mobile-config-firefox";
+    src = fetchFromGitLab {
+      owner = "postmarketOS";
+      repo = "mobile-config-firefox";
+      rev = version;
+      sha256 = "0kqq31h4qdqbn8y65jz27rsn7v85q3bdprdx1fb0c9s3g9lh5zqx";
+    };
+
+    installPhase = ''
+      mkdir $out
+      cp out/home.html $out/
+      cp out/userChrome.css $out/
+      cp src/policies.json $out/
+      cp src/mobile-config-autoconfig.js $out/
+      cp src/mobile-config-prefs.js $out/
+    '';
+
+    fixupPhase = ''
+      sed -i "s|/etc/mobile-config-firefox/userChrome.css|$out/userChrome.css|" $out/mobile-config-autoconfig.js
+      sed -i "s|/usr/share/mobile-config-firefox/home.html|$out/home.html|" $out/policies.json
+    '';
+
+    meta = with lib; {
+      description = "Mobile-friendly Firefox configurations from postmarketOS";
+      homepage = "https://gitlab.com/postmarketOS/mobile-config-firefox";
+      license = licenses.gpl3Plus;
+      maintainers = with maintainers; [ zhaofengli ];
+    };
+  };
+
+  # AutoConfig (forced configs)
+  extraPrefs = builtins.readFile "${mobile-config-firefox}/mobile-config-autoconfig.js";
+
+  extraPolicies = let
+    policies = (builtins.fromJSON (builtins.readFile "${mobile-config-firefox}/policies.json")).policies;
+  in if keepHomepage then lib.filterAttrs (k: v: k != "Homepage") policies else policies;
```

Comment:
Isn't this IFD? I thought nixpkgs and hydra was against that.
Maybe the files could be copied into the tree

Reply:
Sure, let's have a update script that generates a `.nix`.